src/Skip/Skipper.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Core\Skip;
+
+use Rector\Core\Rector\AbstractRector;
+use Symfony\Component\Finder\Finder;
+use Symplify\SmartFileSystem\SmartFileInfo;
+
+final class Skipper
+{
+    /**
+     * @var mixed[]
+     */
+    private $skip = [];
+
+    /**
+     * @param mixed[] $skip
+     */
+    public function __construct(array $skip = [])
+    {
+        $this->skip = $skip;
+    }
+
+    public function shouldSkipFileInfoAndRule(SmartFileInfo $smartFileInfo, AbstractRector $rector): bool
+    {
+        if ($this->skip === []) {
+            return false;
+        }
+        $rectorClass = get_class($rector);
+        if (! array_key_exists($rectorClass, $this->skip)) {
+            return false;
+        }
+
+        $locations = $this->skip[$rectorClass];
+        $filePathName = $smartFileInfo->getPathName();
+
+        foreach ($locations as $location) {
+            if (is_dir($location)) {
+                $finder = new Finder();
+                $finder->files()->in($location)->name('*.php');
+
+                if ($this->isFoundInDirectory($finder, $filePathName)) {
+                    return true;
+                }
+            }
```

Comment:
This method could take a big performance hit on every directory + file + node.

Could you think of better way to check file path is in some directory path?

Reply:
we may can do with : 

```php
$target = $location . DIRECTORY_SEPARATOR . (new SplFileInfo($filePathName))->getFileName();
if (is_file($target == $filePathName)) {
    return true;
}
```

However, this will only work for same structure, not recursive deep inside directory. Do you think that is enough?

Reply:
I've updated code to cache found files in previous rules to static property, if exists, no need to re-search in same directory.